packager/file/http_file.cc

```diff
@@ -297,7 +297,7 @@ void HttpFile::SetupRequest() {
   curl_easy_setopt(curl, CURLOPT_FOLLOWLOCATION, 1L);
   curl_easy_setopt(curl, CURLOPT_WRITEFUNCTION, &CurlWriteCallback);
   curl_easy_setopt(curl, CURLOPT_WRITEDATA,
-                   method_ == HttpMethod::kPut ? nullptr : &download_cache_);
+                   method_ == HttpMethod::kGet ? &download_cache_ : nullptr);
```

Comment:
What's the reason for this change?

Reply:
This change was a part of a [discussion I had with Joey](https://github.com/CaitlinOCallaghan/shaka-packager/pull/3#discussion_r679521915). For background, POST was considered as an upload option when I was fiddling with the LL-DASH implementation; however, the final implementation uses PUT. 

The [POST method is supported by Packager](https://github.com/google/shaka-packager/blob/master/packager/file/http_file.h#L25), but is not currently used by any feature. While looking at the `http_file.cc`, we noticed that this line's original logic did not account for POST. To support POST, which would need the `nullptr` value,  the ternary operator logic was changed.